### DISCUSSION TITLE
Bookmarks are deleted

### DISCUSSION THREAD

**Discussion by ghost** [rubberduck-vba/Rubberduck#5994] [Q&A] [answered]
When Rubberduck 'Indent' feature is used, all bookmarks are deleted. Any fix for this in the future? very annoying.

**Comment by retailcoder** [comment, marked as answer]
The VBIDE API provides exactly zero hooks for bookmarks, so there is exactly zero way of knowing where they are, let alone to manipulate and/or restore them.

Any VBIDE add-in that rewrites code in the editor will have the same issue, including the original Smart Indenter add-in.

I would recommend leveraging the TODO marker feature instead of using VBIDE bookmarks; you can easily configure a list of case-insensitive comment markers (defaults are TODO, NOTE, and BUG) that Rubberduck will recognize as such, and present in an organized list that's much more practical and useful than VBIDE bookmarks.

**Comment by ghost** [comment]
Thank you Mathieu,
I'm an amateur VBA programmer, My go to langue is C#. Now I see what a job your team has embedding your add-in in VBA.
Rubber Duck makes VBA much more pleasant to use, thank you all. I'll take your advice and use the markers; don't know why I didn't try them before.

 Since I'm a novice, I'm not comfortable posting dumb question on Rubber Duck GitHub, Is It ok to send emails to you if I don't abuse the privilege? I'm an old-time programmer now 68 years old, obsolete languages Fortran, Cobol, DEC Diboll, RPG ext.

Dallas Smith
Holland Michigan

Sent from Outloo<http://aka.ms/weboutlook>

________________________________

**Comment by IvenBach** [comment]
It may be intimidating at first to ask but go ahead. We're all learning and your question could become and answer for someone else.

There is a chat room https://chat.stackexchange.com/rooms/14929/vba-rubberducking several members frequent if you're still nervous about posting openly on GH.

Edit: Confidence booster
@Dosman54 My past-self was very intimidated with contributing, but I'm glad I did. An issue I opened just now became an answer that helped someone https://github.com/rubberduck-vba/Rubberduck/issues/5320. Hope this example helps you feel more confident about posting.
